app/design/frontend/Diana/default-theme/web/js/render-stores.js

```diff
@@ -0,0 +1,123 @@
+define([
+    'uiComponent',
+    'jquery',
+    'ko',
+    'underscore',
+    'js/selector-store',
+    'js/selector-city'
+], function(Component, $, ko, _){
+    'use strict';
+
+    return Component.extend({
+        defaults: {
+            currentPageNumber: ko.observable(0),
+            elementsPerPage: 6,
+            quantityOfPages: ko.observable(),
+            quantityOfPagesRenderer: ko.observableArray([]),
+            paginatedElements: ko.observableArray(),
+            selectedCity: ko.observable(),
+            paramPage: ko.observable(),
+            imports: {
+                selectedCity: 'selectorCity:selectedCity',
+                availableStores: 'selectorStore:availableStores'
+            }
+        },
+        initObservable: function () {
+            this._super().observe(['selectedCity', 'availableStores']);
+
+            return this;
+        },
+        initialize: function() {
+            var self = this;
+            this._super();
+            self.setInitialCurrentPageFromURLParameters();
+            self.computeQuantityOfPages(self.availableStores());
+            self.renderPaginatedElements(self.availableStores());
+            self.availableStores.subscribe(function(value) {
+                if (!_.isUndefined(value)) {
+                    self.computeQuantityOfPages(self.availableStores());
+                    self.renderPaginatedElements(self.availableStores());
+                }
+            });
+        },
+        setInitialCurrentPageFromURLParameters: function () {
+            var self = this;
+            var url = new URL(window.location.href);
+            if (!_.isNull(url.href.match(/page=\d/))) {
+                self.paramPage(url.href.match(/page=\d/)[0].match(/\d/)[0]);
+                self.currentPageNumber(self.paramPage() - 1);
+            } else {
+                self.paramPage(null);
+                self.currentPageNumber(0);
+            }
+        },
+        computeQuantityOfPages: function (data) {
+            var self = this;
+            self.quantityOfPagesRenderer([]);
+            self.quantityOfPages = ko.computed(function() {
+                var quantityOfPages = Math.ceil(data.length / self.elementsPerPage);
```

Comment:
@Dianaeva This note can be optimized - `return Math.ceil(data.length / self.elementsPerPage);`